openqa-investigate

```diff
@@ -26,17 +26,27 @@ curl_args=(--user-agent "openqa-investigate")
 echoerr() { echo "$@" >&2; }
 
 clone() {
-    local origin id name_suffix refspec job_data unsupported_cluster_jobs name base_prio clone_settings casedir repo out url clone_id
+    local origin id name_suffix refspec job_data unsupported_cluster_jobs parallel_parents pending_cluster_jobs name base_prio clone_settings casedir repo out url clone_id
     origin=${1:?"Need 'origin'"}
     id=${2:?"Need 'id'"}
     name_suffix=${3+":$3"}
     refspec=${4+$4}
     job_data=$(openqa-cli "${client_args[@]}" --json jobs/"$id")
     # shellcheck disable=SC2181
     [[ $? != 0 ]] && echoerr "unable to query job data for $id: $job_data" && return 1
-    unsupported_cluster_jobs=$(echo "$job_data" | runjq -r '(.job.children["Parallel"] | length) + (.job.parents["Parallel"] | length) + (.job.children["Directly chained"] | length) + (.job.parents["Directly chained"] | length)') || return $?
+
+    # fail on jobs with directly chained dependencies (not supported)
+    unsupported_cluster_jobs=$(echo "$job_data" | runjq -r '(.job.children["Directly chained"] | length) + (.job.parents["Directly chained"] | length)') || return $?
     [[ $unsupported_cluster_jobs != 0 ]] \
-        && echoerr "unable to clone job $id: it is part of a parallel or directly chained cluster (not supported)" && return 2
+        && echoerr "Unable to clone job $id: it is part of a directly chained cluster (not supported)" && return 2
+
+    # fail on parallel children (not supported)
+    # note: We already skip the investigation for parallel children to handle it together with the parent. However, the last good job
+    #       of the parallel parent might be a parallel child so we still need to check whether we have a parallel child.
+    parallel_parents=$(echo "$job_data" | runjq -r '(.job.parents["Parallel"] | length)') || return $?
+    [[ $parallel_parents != 0 ]] \
+        && echoerr "Unable to clone job $id: it is a parallel child; the structure of the parallel cluster might have changed since the last good (not supported)" && return 2
+
```

Comment:
please extract some code as the function "clone" becomes too big

Reply:
I suppose for extending tests that will be necessary anyways :-)